service/s3/s3manager/download.go

```diff
@@ -199,38 +204,48 @@ func (d *downloader) download() (n int64, err error) {
 // of bytes to be read so that the worker manager knows when it is finished.
 func (d *downloader) downloadPart(ch chan dlchunk) {
 	defer d.wg.Done()
-
 	for {
 		chunk, ok := <-ch
 		if !ok {
 			break
 		}
+		d.downloadChunk(chunk)
+	}
+}
 
-		if d.geterr() == nil {
-			// Get the next byte range of data
-			in := &s3.GetObjectInput{}
-			awsutil.Copy(in, d.in)
-			rng := fmt.Sprintf("bytes=%d-%d",
-				chunk.start, chunk.start+chunk.size-1)
-			in.Range = &rng
+// getChunk grabs a chunk of data from the body.
+func (d *downloader) getChunk() {
```

Comment:
Might make sense to move this logic into the `download` function. Being its own method accessing data like this could introduce data races if this method is called somewhere else unexpectedly.